src/jackdaw/serdes/resolver.clj

```diff
@@ -0,0 +1,50 @@
+(ns jackdaw.serdes.resolver
+  (:require
+    [jackdaw.serdes.avro :as avro-serde]
+    [jackdaw.serdes.json :as json-serde]))
+
+(defn local-schema-loader []
+  (fn [topic-config key?]
+    (if key?
+      "{\"type\":\"string\"}"
+      (when-let [r (.getResource
+                     (.getContextClassLoader (Thread/currentThread))
+                     (str (:value-schema-name topic-config) ".json"))]
+        (slurp r)))))
+
+(defn create-avro-serde [schema-registry-url schema-loader topic-name key?]
+  (avro-serde/serde
+    ;; Registry of avro <-> clj types
+    (merge avro-serde/+base-schema-type-registry+ avro-serde/+UUID-type-registry+)
+    ;; Schema registry config
+    {:avro.schema-registry/url schema-registry-url}
+    ;; Avro serdes config
+    {:key? key?
+     :avro/schema (schema-loader topic-name key?)}))
+
+(defn create-serdes-lookup [schema-registry-url schema-loader]
+  {:avro-key (fn [topic-config]
+               (create-avro-serde
+                 schema-registry-url schema-loader topic-config true))
+   :avro-value (fn [topic-config]
+                 (create-avro-serde
+                   schema-registry-url schema-loader topic-config false))
+   :json (fn [_]
+           (json-serde/serde))
+   :string (fn [_]
+             (org.apache.kafka.common.serialization.Serdes/String))})
+
+(defn serdes-resolver [schema-registry-url schema-loader]
+  (fn [topic-config]
+    (letfn [(get-serde [k]
+              (let [f (get
+                        (create-serdes-lookup schema-registry-url schema-loader)
+                        (get topic-config k))]
+                (f topic-config)))]
+      (merge
+        topic-config
+        {:key-serde (get-serde :key-serde)
+         :value-serde (get-serde :value-serde)}))))
+
+(defn local-serdes-resolver [schema-registry-url]
+  (serdes-resolver schema-registry-url (local-schema-loader)))
```

Comment:
Add a `remote-series-resolver` which takes the schema to use form the registry rather than local disk.